src/components/Karma/utils.ts

```diff
@@ -0,0 +1,21 @@
+export const renderCircleColor = (karma: number) => {
+  if (karma > 87.5) {
+    return ['#50BA57', '#C7EECA', '#50BA57'];
+  }
+  if (karma > 75) {
+    return ['#70BF21', '#B3E57F', '#70BF21'];
+  }
+  if (karma > 62.5) {
+    return ['#A0C527', '#D1EA82', '#A0C527'];
+  }
+  if (karma > 50) {
+    return ['#D3D20C', '#F3F370', '#D3D20C'];
+  }
+  if (karma > 37.5) {
+    return ['#E4B92E', '#FFE591', '#E4B92E'];
+  }
+  if (karma > 25) {
+    return ['#F4833C', '#F9BE9A', '#F4833C'];
+  }
+  return ['#DF001F', '#F5B0B9', '#DF001F'];
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderCircleColor` function could be optimized by using an array of thresholds and colors instead of multiple if statements. This would make the code more maintainable, as adding or removing a color range would only require modifying the array, not adding or removing an if statement.

Here's how you could refactor this:

```diff
- export const renderCircleColor = (karma: number) => {
-   if (karma > 87.5) {
-     return ['#50BA57', '#C7EECA', '#50BA57'];
-   }
-   if (karma > 75) {
-     return ['#70BF21', '#B3E57F', '#70BF21'];
-   }
-   if (karma > 62.5) {
-     return ['#A0C527', '#D1EA82', '#A0C527'];
-   }
-   if (karma > 50) {
-     return ['#D3D20C', '#F3F370', '#D3D20C'];
-   }
-   if (karma > 37.5) {
-     return ['#E4B92E', '#FFE591', '#E4B92E'];
-   }
-   if (karma > 25) {
-     return ['#F4833C', '#F9BE9A', '#F4833C'];
-   }
-   return ['#DF001F', '#F5B0B9', '#DF001F'];
- };

+ export const renderCircleColor = (karma: number) => {
+   const colorRanges = [
+     { threshold: 87.5, colors: ['#50BA57', '#C7EECA', '#50BA57'] },
+     { threshold: 75, colors: ['#70BF21', '#B3E57F', '#70BF21'] },
+     { threshold: 62.5, colors: ['#A0C527', '#D1EA82', '#A0C527'] },
+     { threshold: 50, colors: ['#D3D20C', '#F3F370', '#D3D20C'] },
+     { threshold: 37.5, colors: ['#E4B92E', '#FFE591', '#E4B92E'] },
+     { threshold: 25, colors: ['#F4833C', '#F9BE9A', '#F4833C'] },
+     { threshold: 0, colors: ['#DF001F', '#F5B0B9', '#DF001F'] }
+   ];
+
+   for (let i = 0; i < colorRanges.length; i++) {
+     if (karma > colorRanges[i].threshold) {
+       return colorRanges[i].colors;
+     }
+   }
+ };
```
This refactored function iterates over the `colorRanges` array and returns the color array for the first range where the karma score is greater than the threshold. This behavior is equivalent to the original function.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
ignore, will implement later on v2